enacts/calc.py

```diff
@@ -65,28 +69,29 @@ def water_balance(
     Examples
     --------
     """
-    # Get time_coord info
-    time_coord_size = daily_rain[time_coord].size
-    # Get all the rain-et deltas:
-    delta_rain_et = daily_rain - et
-    # Intializing sm
-    soil_moisture = xr.DataArray(
-        data=np.empty(daily_rain.shape),
-        dims=daily_rain.dims,
-        coords=daily_rain.coords,
-        name="soil_moisture",
-        attrs=dict(description="Soil Moisture", units="mm"),
+    # Initializing soil_moisture
+    soil_moisture = water_balance_step(
+        sminit,
+        daily_rain.isel({time_dim: 0}).expand_dims(dim=time_dim),
+        et,
+        taw,
     )
-    soil_moisture[{time_coord: 0}] = (
-        sminit + delta_rain_et.isel({time_coord: 0})
-    ).clip(0, taw)
-    # Looping on time_coord
-    for i in range(1, time_coord_size):
-        soil_moisture[{time_coord: i}] = (
-            soil_moisture.isel({time_coord: i - 1})
-            + delta_rain_et.isel({time_coord: i})
-        ).clip(0, taw)
-    water_balance = xr.Dataset().merge(soil_moisture)
+    if not reduce:
+        soil_moisture, _ = xr.align(soil_moisture, daily_rain[time_dim], join="outer")
```

Comment:
I guess the thing being assigned to `_` is `daily_rain` broadcast to the dimensions of `soil_moisture`. Seems like a waste to allocate that big array only to throw it away. Could this be done with `broadcast_like` instead?

Reply:
(Since this happens only once, it's not a big deal. )

Reply:
broadcast_like seems even more straightforward.